validate.go

```diff
@@ -71,3 +71,34 @@ func ComposeValidators(validators ...Validator) Validator {
 		return nil
 	}
 }
+
+// isZero returns true if the passed value is the zero object
+func isZero(v reflect.Value) bool {
```

Comment:
Even if they both work, I like your solution more since way fewer lines of code - I'll change it now and make sure the tests pass (i'm sure they will).